### PR TITLE
 commit 02.02.2019

### PR DESCRIPTION
**Pull request by MateAcademy**

### CODE REVIEW

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 1]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
...
+
+	@Override
+	@DELETE
+	@Path("/hrs/{surname}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response removeHR(@PathParam("groupId")int groupId, @PathParam("surname")String surname) {
+		MateGroup mateGroup = mateGroups.get(groupId);
+		if (mateGroup != null) {
+			mateGroup.getHumanResources().removeIf(f -> f.getSurname().equals(surname));
+			return Response.status(Status.OK).entity(mateGroups.get(groupId).getHumanResources())
```
Comment (Cherkashyn) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java]: U can use just just mateGroup instead mateGroups.get(groupId).

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 2]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
...
+		if (mateGroup != null) {
+			mateGroup.getHumanResources().add(newHR);
+			return Response.status(ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
+		}
+		return Response.status(Status.NOT_FOUND).build();
+	}
+
+	@Override
+	@DELETE
+	@Path("/hrs/{surname}")
```
Comment (Cherkashyn) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java:103]: You use different pathes in methods. In previous POST you used "/humanResources", in this method you use just "/hrs".
It's critical.

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 3]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
...
+	@PUT
+	@Path("/hrs/{surname}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response updateStartWorkYearHR(@PathParam("groupId") int groupId, @PathParam("surname") String surname,
+										  @QueryParam("newStartWorkYear") int newStartWorkYear) {
+		MateGroup mateGroup = mateGroups.get(groupId);
+		if (mateGroup != null) {
+			mateGroup.getHumanResources().stream().filter((f -> f.getSurname().equals(surname)))
+					.collect(Collectors.toList()).forEach(f -> f.setStartWorkYear(newStartWorkYear));
+			return Response.status(Status.OK).entity(mateGroups.get(groupId).getHumanResources())
```
Comment (Cherkashyn) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java]: Same. Double usage of mateGroups.get(groupId)

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 4]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
...
+					.type(MediaType.APPLICATION_JSON).build();
+		}
+		return Response.status(Status.NOT_FOUND).build();
+	}
+
+	@Override
+	@PUT
+	@Path("/hrs/{surname}")
+	@Produces(MediaType.APPLICATION_JSON)
```
Comment (Cherkashyn) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java]: If you return entity, you have to use @Consumes also

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 5]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
...
+
+	@Override
+	@GET
+	@Path("/hrs/{surname}")
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response showSingleHR(@PathParam("groupId") int groupId, @PathParam("surname") String surname) {
+		MateGroup mateGroup = mateGroups.get(groupId);
+		if (mateGroup != null) {
+			return Response.status(Status.OK)
+					.entity(mateGroups.get(groupId).getHumanResources().stream()
```
Comment (Cherkashyn) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java:152]: It's hard to read long line of code in return line.

**Review by Cherkashyn — CHANGES_REQUESTED** [review]
Good, but update according to comments.

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java (hunk 1 above)
Comment (MateAcademy) [reply src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java]: ok

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java (hunk 2 above)
Comment (MateAcademy) [reply src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java:103]: I dont agree with you that it is critical, I try humanResources or /hrs -my programm  work

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 6]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
 
 		return Response.status(ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
 	}
+
+	@Override
+	@PUT
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java:87]: better post

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 7]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
...
+	@Path("/humanResources")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response addHR(@PathParam("groupId")int groupId, HumanResource newHR) {
+
+		MateGroup mateGroup = mateGroups.get(groupId);
+		if (mateGroup != null) {
+			mateGroup.getHumanResources().add(newHR);
+			return Response.status(ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java:96]: better return only code

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java (hunk 2 above)
Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java:103]: It's critical. You work with resources. Hard understanding your's clients why you did that

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java (hunk 1 above)
Comment (SerhiiPasko) [reply src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java]: "Ok". Where is changes?

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

File: src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java [hunk 8]
```diff
@@ -80,4 +82,77 @@ public Response removeStudent(@PathParam("groupId")int groupId, @PathParam("surn
 
 		return Response.status(ACCEPTED).entity(mateGroup).type(MediaType.APPLICATION_JSON).build();
 	}
+
+	@Override
+	@POST
+	@Path("/humanResources")
```
Comment (SerhiiPasko) [inline src/main/java/com/mateacademy/lessons13/rest/MateGroupServiceImpl.java]: ????????????????

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments

**Pull request merged**